dbms/CMakeLists.txt

```diff
@@ -414,6 +414,16 @@ endif()
 if (USE_JEMALLOC)
     dbms_target_include_directories (SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # used in Interpreters/AsynchronousMetrics.cpp
     target_include_directories (clickhouse_common_io SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # new_delete.cpp
+    # common/memory.h
+    if (MAKE_STATIC_LIBRARIES OR NOT SPLIT_SHARED_LIBRARIES)
+        # skip if we have bundled build, since jemalloc is static in this case
+    elseif (${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$")
```

Comment:
this test doesn't work at all. `JEMALLOC_LIBRARIES` always equals to `jemalloc`. Need to find another way of determining the actually library type.

Reply:
>this test doesn't work at all

It works for unbundled build.

jemalloc can be:
- bundled - `jemalloc`, always static
- static - `/path/to/libjemalloc.a` 
- shared - `/path/to/libjemalloc.so`

So we can fix this by using `(NOT "${JEMALLOC_LIBRARIES}" MATCHES 
 "${CMAKE_SHARED_LIBRARY_SUFFIX}")`, agree?

Smth like below:

```patch
diff --git a/dbms/CMakeLists.txt b/dbms/CMakeLists.txt
index 94f82d230a..038824527c 100644
--- a/dbms/CMakeLists.txt
+++ b/dbms/CMakeLists.txt
@@ -418,8 +418,8 @@ if (USE_JEMALLOC)
     target_include_directories (clickhouse_common_io SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # new_delete.cpp
     # common/memory.h
     if (MAKE_STATIC_LIBRARIES OR NOT SPLIT_SHARED_LIBRARIES)
-        # skip if we have bundled build, since jemalloc is static in this case
-    elseif (${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$")
+        # skip if clickhouse_common_io is static
+    elseif (NOT "${JEMALLOC_LIBRARIES}" MATCHES "${CMAKE_SHARED_LIBRARY_SUFFIX}$")
         # if the library is static we do not need to link with it,
         # since in this case it will be in libs/libcommon,
         # and we do not want to link with jemalloc multiple times.
```